### PR TITLE
Fix notebook sync

### PR DESCRIPTION
**Pull request by akhenry**
Fixes https://github.com/nasa/openmct/issues/4737

### Description
1. [Removes unnecessary re-fetching of notebook](https://github.com/nasa/openmct/pull/4738/files#diff-65eecdb132430c1ee14c1f6be3551e25a5e64cec0b7370c7d19b47472101d259L643) on every new entry. This was sometimes preventing users from creating new entries due to [changes in the Object API](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L193) that causes all local models of an object to be replaced with the server model on `get`. This meant that local notebook entries would in some cases be immediately replaced with a stale model from the server.
2. Fixes a bug where [object observe listeners were being removed too eagerly](https://github.com/nasa/openmct/pull/4738/files#diff-95c2adc3908b33dfe8ce5d440783f276110e9a45768f5e45a870e1c86c2c3ea6R25), meaning that objects would stop listening for remote changes.
3. Fixes another [bug in the notebook conflict resolution code](https://github.com/nasa/openmct/pull/4738/files#diff-95c2adc3908b33dfe8ce5d440783f276110e9a45768f5e45a870e1c86c2c3ea6R34) where local changes to a notebook would be lost, also due to changes in the Object API](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L193) that cause all local models of an object to be replaced with the server model on `get`. Code will now make a local copy of notebook entries before fetching the server version of the object, so that local changes do not get overwritten.
4. [Removes duplicate `refresh` method](https://github.com/nasa/openmct/pull/4738/files#diff-7b3a993b89a077474a9a4cf7fca79c18131e8e222951d164d2848ab147afbd24L473) in Object API. This was not causing any issues, it's just some code cleanup. Duplicate was caused by the same change being made in two different PRs.

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [ ] Unit tests included and/or updated with changes? No functional changes.
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### CODE REVIEW

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
Nice investigating!
One change to put objectAPI.refresh back.

File: src/api/objects/ObjectAPI.js [hunk 1]
```diff
@@ -465,23 +465,6 @@ ObjectAPI.prototype.mutate = function (domainObject, path, value) {
     }
 };
 
-/**
```
[the remark this answers is not in the document]
Comment (akhenry) [reply src/api/objects/ObjectAPI.js:468]: I'm not actually removing it, the function is duplicated [here](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L473) and [here](https://github.com/nasa/openmct/blob/master/src/api/objects/ObjectAPI.js#L525). I'm just removing one of the copies.

**Comment by shefalijoshi** [comment]
@akhenry Please resolve confict so I can merge this. Thanks!

**Review by shefalijoshi — APPROVED** [review]

**Comment by shefalijoshi** [comment]
### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled `bug`? (only applicable if this PR is for a bug fix)
* [ ] List of Acceptance Tests Performed.

**Pull request merged**